### DISCUSSION TITLE
Google Gemini Support for pandasAi

### DISCUSSION THREAD

**Discussion by ulloogeo** [sinaptik-ai/pandas-ai#986] [Ideas]
Hey Folks,
Any plans to support Google Gemini any time soon with pandasAi?

Thx

**Comment by ulloogeo** [comment]
I guess I asked too soon :-)
Looked at the pandas-ai source code and found this under the llm folder

`from pandasai.llm.google_gemini import GoogleGemini`

But, it does not work. Throws this error:

`ModuleNotFoundError: No module named 'pandasai.llm.google_gemini'`

Any ideas how we can use this class?

**Reply by mspronesti** [reply]
There's a bug in the current version of pandas-ai, that I've patched in #980.

Please follow the instructions mentioned in this [comment](https://github.com/Sinaptik-AI/pandas-ai/issues/979#issuecomment-1974764566) and let me know if that solves your issue :)

**Comment by ulloogeo** [comment]
@mspronesti Here is what is happening:

- First I updated pandasai to 2.0.2 
- Then I ran the the following code which runs fine:

> from pandasai import SmartDataframe
> from pandasai.llm.google_gemini import GoogleGemini
> 
> llm = GoogleGemini(api_key="***********************************")
> 
> df = SmartDataframe("/home/001/Documents/NLP Projects/PandasAI/wedding expenses.csv", config={"llm": llm})
> response = df.chat("Waht was the total of all transactions?")
> print(response)

- When I run the following code it gives me the `TypeError: _generate_text() takes 2 positional arguments but 3 were given`. Note that only the prompt has been changed.

> from pandasai import SmartDataframe
> from pandasai.llm.google_gemini import GoogleGemini
> 
> llm = GoogleGemini(api_key="***********************************")
> 
> df = SmartDataframe("/home/001/Documents/NLP Projects/PandasAI/wedding expenses.csv", config={"llm": llm})
> response = df.chat("Waht was the total of top 5 transactions?")
> print(response)

Because I am new to GitHub I am not sure what to do next to try out the fix you recommended.

Thanks

**Reply by mspronesti** [reply]
Just run the commands mentioned [here](https://github.com/Sinaptik-AI/pandas-ai/issues/979#issuecomment-1974764566), and then run your script again :)
